src/modules/cspec/geometry/CspecGeometry.h

```diff
@@ -0,0 +1,59 @@
+// Copyright (C) 2022 European Spallation Source, ERIC. See LICENSE file
+//===----------------------------------------------------------------------===//
+///
+/// \file
+///
+/// \brief Calculate pixelid from tube and amplitudes
+///
+//===----------------------------------------------------------------------===//
+
+#pragma once
+
+#include <common/debug/Trace.h>
+#include <logical_geometry/ESSGeometry.h>
+#include <modules/caen/geometry/Geometry.h>
+#include <string>
+#include <vector>
+//
+// #undef TRC_LEVEL
+// #define TRC_LEVEL TRC_L_DEB
+
+namespace Caen {
+class CspecGeometry : public Geometry {
+public:
+    CspecGeometry(Config &CaenConfiguration);
+  uint32_t calcPixel(DataParser::CaenReadout &Data);
+  bool validateData(DataParser::CaenReadout &Data);
+
+  int PackTubes{24}; // tubes per triplet (might be obvious from the name)
+
+  /// \brief return the global x-offset for the given identifiers
+  int xOffset(int Ring, int Tube);
+
+  /// \brief return the global y-offset for the given identifiers
+  int yOffset(int Tube);
+
+  /// \brief return local x-coordinate from amplitudes
+  int xCoord(int AmpA, int AmpB) {
+    int Coord = reverse(posAlongTube(AmpA, AmpB)) % (NPos / 3);
+    XTRACE(DATA, DEB, "AmpA %d, AmpB %d, xCoord %d", AmpA, AmpB, Coord);
+    return Coord;
+  }
+
+  /// \brief return local y-coordinate from amplitudes
+  int yCoord(int AmpA, int AmpB) {
+    return (reverse(posAlongTube(AmpA, AmpB)) * PackTubes) / NPos;
```

Comment:
there seem to be some assumptions going on here that I don't know about.

But given the height of CSPEC I would expect only a single tube per (A,B) pair?
If this is the case then the position calculations are wrong. If the assumption is wrong and 
you have inside information we need to update the relevant ICD's accoringly.

Reply:
The assumptions are all BIFROST based (I copied that directory as a base and didn't modify the geometry information sufficiently). The latest commit 'fixes' this by punting on most of it, and assuming that the detector will be cylindrical with all tubes at constant radius and height and all the same length.

Yes, it's very likely that every tube will give a single (A,B) pair.